soc/silabs/silabs_s2/xg23/Kconfig.soc

```diff
@@ -13,12 +14,24 @@ config SOC_SERIES_EFR32ZG23
 	help
 	  Silicon Labs EFR32ZG23 Series MCU
 
+config SOC_SERIES_EFM32PG23
+	bool
+	select SOC_SILABS_XG23
+	help
+	  Silicon Labs EFM32PG23 Series MCU
+
 config SOC_PART_NUMBER_EFR32ZG23B020F512IM48
 	bool
 	select SOC_SERIES_EFR32ZG23
 
+config SOC_PART_NUMBER_EFM32PG23B310F512IM48
```

Comment:
This should be `SOC_EFM32PG23B310F512IM48`, from the previous discussion. Over time, all the `PART_NUMBER` infixes should be removed from all the SoC selection options, but this PR only needs to do it for the new ones that are being introduced.